### ISSUE TITLE
Configurable Product option is still displayed and available on Product view page after Place Order with full option qty

### ISSUE THREAD

**Issue by zakdma** [magento/inventory#3102]
### Preconditions (*)

1. Magento 2.4-develop Inventory 1.2-develop

### Steps to reproduce (*)

1. Create Configurable product with two options with qty of 10
2. Place order with any of the options with full of qty (10)
3. View the Configurable product on Product view page.

### Expected result (*)

4. Product with option that was ordered with full qty should be out of stock (option from the select should be disabled)

### Actual result (*)

4. Product with option that was ordered with full qty is not disabled.

### Additional information
5. If to open the Configurable product fron an anonymous browser tab, the option will be enabled AS NOT expected.

The below steps are present for help to clarify what is going on. 

### Steps to reproduce 2
1. Create Configurable product with two options with qty of 10
2. Login as a customer.
3. Place order with any of the options with fill qty (10)
4. View the Configurable product on Product view page.

### Expected result 2
5. Product with option that was ordered with full qty should be out of stock (option from the select should be disabled)

### Actual result 2
5. Product with option that was ordered with full qty is not disabled.

### Additional information 2
6. If to open the Configurable product fron an anonymous browser tab, the option will be disabled AS expected.
or
6. If to logout the customer, the option will be disabled AS expected.

In summary 
It is reproducible even with Simple product, and no matter which stock to use (legacy or custom)

Related to MC-34328 internal ticket

**Comment by IbrahimS2** [comment]
Same problem here, it should check against Salable Quantity rather checking against Qty itself, thus causing a lot of misleading and costing businesses a lot of money!

**Comment by thai2301** [comment]
up!

**Comment by engcom-November** [comment]
Hello,

Thank you for your report and collaboration.

We attempted to reproduce this issue on the 2.4-develop branch, and the issue was not reproducible.

Could you please verify the behavior on the latest version of Magento and let us know if the issue still persists?
If you continue to experience any problems, please feel free to reach out for further assistance.

Hence marking this issue as "Needs Update"

Thank you!

**Comment by engcom-November** [comment]
Hello,

This issue is being closed since it has not been updated in a long time. Please feel free to reopen or raise a new ticket if the issue still exists.

Thank You!
